Jenkinsfile

```diff
@@ -0,0 +1,207 @@
+ *
+ * 5. Neither the name of Intel Corporation, nor the name of any Contributor
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+// To use a test branch (i.e. PR) until it lands to master
+// I.e. for testing library changes
+//@Library(value="pipeline-lib@your_branch") _
+
+pipeline {
+    agent none
+
+    stages {
+        stage('Cancel Previous Builds') {
+            when { changeRequest() }
+            steps {
+                cancelPreviousBuilds()
+            }
+        }
+        stage('Lint') {
+            stages {
+                stage('RPM Lint') {
+                    agent {
+                        dockerfile {
+                            filename 'Dockerfile.centos.7'
+                            label 'docker_runner'
+                            additionalBuildArgs  '--build-arg UID=$(id -u)'
+                            args  '--group-add mock --cap-add=SYS_ADMIN --privileged=true'
+                        }
+                    }
+                    steps {
+                        // slurm.spec fails lint.
+                        sh script: '''cp -f slurm.spec_centos slurm.spec
+                                      make rpmlint''',
+                           returnStatus: true
+                    }
+                }
+            }
+        }
+        stage('Build') {
+            parallel {
+                stage('Build on CentOS 7') {
+                    agent {
+                        dockerfile {
+                            filename 'Dockerfile.centos.7'
+                            label 'docker_runner'
+                            args '--group-add mock' +
+                                 ' --cap-add=SYS_ADMIN' +
+                                 ' --privileged=true'
+                            additionalBuildArgs '--build-arg UID=$(id -u)'
+                        }
+                    }
+                    steps {
+                        sh '''rm -rf artifacts/centos7/
+                              mkdir -p artifacts/centos7/
+                              cp -f slurm.spec_centos slurm.spec
```

Comment:
This should be able to go into `Makefile_packaging.mk`, shouldn't it?  Generally speaking, the goal of a `make` target should be to do **everything** necessary to create the target **without** having to do any prep for the target.  Otherwise we're doing some of `make`'s work for it, and requiring the user to gain and have some "build knowledge" around making the target.

Reply:
Makefile_packaging.mk expects the spec file to exist before it starts evaluating any rules.

I could not get the %include to work in the build chroot environment to use that method of selecting.

So since the SLES-12SP3 is eventually going away, this seemed like the best compromise until then.